core/view/src/parser/index.ts

```diff
@@ -146,36 +77,54 @@ export class Parser {
     type: Node.ChildrenTypes = NodeType.Value,
     options: ParseObjectOptions = { templateDepth: 0 }
   ): Node.Node | null {
-    if (hasApplicability(obj)) {
-      return this.parseApplicability(obj, type, options);
+    const nodeType = this.hooks.determineNodeType.call(obj);
+
+    if (nodeType !== undefined) {
+      const parsedNode = this.hooks.parseNode.call(
+        obj,
+        type,
+        options,
+        nodeType
+      );
+      if (parsedNode) {
+        return parsedNode;
+      }
     }
 
-    if (hasSwitch(obj)) {
-      return this.parseSwitch(obj, options);
-    }
+    interface Obj {
+      children: Node.Child[];
 
-    let value: any;
-    const children: Node.Child[] = [];
+      value: any;
+    }
 
-    /** Parse a nested child and add it to the parent */
-    const parseLocalObject = (objToParse: unknown, path: string[] = []) => {
+    const parseLocalObject = (
+      currentValue: any,
+      objToParse: unknown,
```

Comment:
What's the difference between `currentValue` and `objToParse`?

Reply:
there are cases when the current value is just a value with nothing to parse, so it doesnt contain a children element on the return